rholang/src/main/scala/coop/rchain/rholang/interpreter/PrettyPrinter.scala

```diff
@@ -174,16 +208,32 @@ case class PrettyPrinter(freeShift: Int, boundShift: Int) {
   private def buildPattern(patterns: Seq[Channel]): (Int, String) =
     ((0, "") /: patterns.zipWithIndex) {
       case ((patternsFree, string), (pattern, i)) =>
-        (patternsFree + freeCount(pattern), string + buildString(pattern) + {
-          if (i != patterns.length - 1) ", "
-          else ""
-        })
+        (patternsFree + freeCount(pattern),
+          string +
+            this
+              .copy(
+                boundShift = 0,
+                boundId = setBoundId(),
```

Comment:
Why are you changing boundId here? you should change it once for all the patterns, and leave it.